Packages/PrimitivesComponents/Sources/ViewModels/InputValidationViewModel.swift

```diff
@@ -0,0 +1,90 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import Foundation
+
+public enum InputValidationMode {
+    case live, manual
+}
+
+@MainActor
+@Observable
+public final class InputValidationViewModel {
+    public var text: String = "" {
+        didSet {
+            if text.isEmpty {
+                error = nil
+            } else if mode == .live {
+                validate()
+            }
+        }
+    }
+
+    public private(set) var error: (any LocalizedError)?
+
+    private var mode: InputValidationMode
+    private var validators: [any TextValidator]
+
+    public init(
+        mode: InputValidationMode = .manual,
+        validators: [any TextValidator]
+    ) {
+        self.mode = mode
+        self.validators = validators
+    }
+
+    public var isValid: Bool { error == nil && !text.isEmpty }
+}
+
+// MARK: - Public
+
+extension InputValidationViewModel {
+
+    @discardableResult
+    public func validate() -> Bool {
+        error = Self.validate(text, with: validators)
+        return error == nil
+    }
+
+    public func updateValidators(_ validators: [any TextValidator]) {
```

Comment:
For consistency with other methods
```suggestion
    public func update(validators: [any TextValidator]) {
```